libs/base/Data/List.idr

```diff
@@ -358,6 +358,20 @@ export
 zip3 : List a -> List b -> List c -> List (a, b, c)
 zip3 = zipWith3 \x, y, z => (x, y, z)
 
+||| Split a list of pairs into two lists
+export
+unzip : List (a, b) -> (List a, List b)
```

Comment:
Can we have `unzipWith : (a -> (b, c)) -> List a -> (List b, List c)` too? :pleading_face: 
I am always annoyed that we have `zipWith`, `zip`, `unzip` but no `unzipWith`.

Reply:
Also, if we have `zip3`, we should probably also have `unzip3` and `unzipWith3`.

Reply:
> Can we have `unzipWith : (a -> (b, c)) -> List a -> (List b, List c)` too? 🥺
> I am always annoyed that we have `zipWith`, `zip`, `unzip` but no `unzipWith`.

Good idea.  I will add them.

Reply:
Just one more note: If we add `unzipWith`, we should have this also for `Vect`, `Stream`, and `List1` (probably also for `Maybe` and `Either e`) to be consistent across different (un)zippable data types. I wonder, whether we actually should have an interface for this (like MonadZip in Haskell). Probably this should not all go in this PR, though?

Reply:
Added `zipWith` and `zipWith3`